### PR TITLE
Added aliases to the video

### LINKED ISSUE #659: Make aliases available to be used by providers
This is a **feature request** to make aliases information available to be used by providers.

The [use case](https://github.com/pymedusa/SickRage/issues/505) is `Marverl's Agents of S.H.I.E.L.D` in LegendasTv provider.

Legendas.TV website data is not consistent. Some Marvel's series doesn't have `Marvel's` in their name, others do. Some of them use `Marvels`, others use `Marvel's`.

MAoS can only be searched using its alias: `Agents of S.H.I.E.L.D`
http://legendas.tv/legenda/sugestao/agents%20of%20s.h.i.e.l.d
Using its official name `Marvel's Agents of S.H.I.E.L.D` returns no result
http://legendas.tv/legenda/sugestao/marvel's%20agents%20of%20s.h.i.e.l.d

But `Marvel's Jessica Jones` can be search using the official name:
http://legendas.tv/legenda/sugestao/marvel's%20jessica%20jones

My suggestion is to enrich the video object with aliases information when running tvdb refiner, so any provider can make use of them when needed (e.g.: when no series search result is found)
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/33973777-make-aliases-available-to-be-used-by-providers?utm_campaign=plugin&utm_content=tracker%2F321252&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F321252&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ratoaq2**
Proposal to fix #659.
- Added aliases to the video
- Tvdb refiner to use it
-  LegendasTV search fallback to use aliases.
- Added test scenario: `Turn: Washington's Spies` can only be found because of its alias: `Turn`

### CODE REVIEW

File: tests/cassettes/legendastv/test_list_subtitles_episode_alternative_series.yaml [hunk 1]
```diff
@@ -0,0 +1,5985 @@
...
+- request:
+    body: null
+    headers:
+      Accept: ['*/*']
+      Accept-Encoding: ['gzip, deflate']
+      Connection: [keep-alive]
+      Cookie: [PHPSESSID=8ae41jk8qo4omid6gkmduu9dg3; au=3673290---198d5f2e1a202d0c2878ae871f911763d7cd76c0]
+      User-Agent: [Subliminal/2.1]
+    method: GET
+    uri: http://legendas.tv/util/carrega_legendas_busca_filme/44435/1/-/1
```
Comment (fernandog) [inline tests/cassettes/legendastv/test_list_subtitles_episode_alternative_series.yaml]: @ratoaq2 can you rebase and use this URI please: http://legendas.tv/legenda/busca/-/1/-/0/44435
thanks

**Comment by Diaoul** [comment]
Can you update other providers to use this as well?

**Comment by ratoaq2** [comment]
@Diaoul, I'll need to find some free time to do that. Probably in a week or so.

**Comment by fernandog** [comment]
@ratoaq2 I added to the remaining providers. Feel free to cherry-pick

https://github.com/fernandog/subliminal/commit/cc064e2e30fac0278a18083e65d056d2ad9b786c

pytests passed:
tests/test_addic7ed.py::test_list_subtitles_episode_alternative_series PASSED
tests/test_opensubtitles.py::test_list_subtitles_episode_alternative_series PASSED
tests/test_podnapisi.py::test_list_subtitles_episode_alternative_series PASSED
tests/test_tvsubtitles.py::test_list_subtitles_episode_alternative_series PASSED

For providers: napiprojekt, thesubdb, shooter there is only hash match
Provider subscenter was ignored as its currently not working

**Comment by ratoaq2** [comment]
Did the other providers got different search results  because of the aliases? Did the aliases make a difference for them?

**Comment by fernandog** [comment]
For TVsubtitles and Addi7ed it only found because of the alternative title

```
        with Addic7edProvider() as provider:
            subtitles = provider.list_subtitles(video, languages)
>       assert {subtitle.download_link for subtitle in subtitles} == expected_subtitles
E       AssertionError: assert set([]) == set(['updated/1/125243/0', 'updated/1/125243/1'])
E         Extra items in the right set:
E         'updated/1/125243/0'
E         'updated/1/125243/1'
```

```
        with TVsubtitlesProvider() as provider:
            subtitles = provider.list_subtitles(video, languages)
>       assert {subtitle.subtitle_id for subtitle in subtitles} == expected_subtitles
E       assert set([]) == set([307588])
E         Extra items in the right set:
E         307588
```

But for opensubtitles and podnapisi it found without the alternative title

**Comment by ratoaq2** [comment]
For opensubtitles and podnasi, do you know if there's a movie or show which will only return results because of the alternative titles?

**Comment by fernandog** [comment]
I tried some other shows and it always find. I can keep searching

**Comment by Diaoul** [comment]
It'd be great to find a show that only yield results thanks to the alternative names. Even as a fallback. This is possibly hard to find with no related issue...

**Comment by fernandog** [comment]
ha! found one for podnapisi. commited alread (fixed up)

File: subliminal/providers/addic7ed.py [hunk 2]
```diff
@@ -272,8 +273,14 @@ def query(self, series, season, year=None, country=None):
         return subtitles
 
     def list_subtitles(self, video, languages):
-        return [s for s in self.query(video.series, video.season, video.year)
-                if s.language in languages and s.episode == video.episode]
+        titles = [video.title] + video.alternative_series
```
Comment (Diaoul) [inline subliminal/providers/addic7ed.py]: Isn't it `video.series` instead of `video.title`?

Comment (fernandog) [reply subliminal/providers/addic7ed.py]: yes

@ratoaq2 can you fix it please?

**Pull request merged**